### ISSUE TITLE
lower accuracy when pretrained word embedding is used

### ISSUE THREAD

**Issue by mrpega** [OpenNMT/OpenNMT-py#1477]
Does anyone encounter this problem? It doesn't make any sense that well initialised word embedding does not perform as well as normal xavier init method.

I have tried glove 100, 200, and 300 dimension word embedding vectors.

I'm using the transformer architecture, 512-dim hidden state. I suppose the X-dim are initialised from word embedding and rest are Xavier-initialised.

Thanks all!

**Comment by rishibommasani** [comment]
Without specifying a task/training duration/some set of hyperparameters, its hard to say whether it makes sense or doesn't for pretrained GloVe embeddings to be outperformed by randomly initialized embeddings. What are you even comparing?
Is it frozen or warm GloVe embeddings you are comparing and are you comparing equal dimensional random initializations to GloVe embeddings?
I have definitely seen cases (and there are papers that verify this empirically, though they are comparatively few compared to the reverse direction) where randomly initialized embeddings outperform GloVe embeddings of the same size. 
Perhaps orthogonally, xavier initialization has been shown to be reasonably important for Transformer models.

**Comment by alphadl** [comment]
initialized or just fixed with pre-trained word vector may be suitable for classification problem or general NLU tasks (e.g., Labeling, Parsing, Classification, Answer selection for QA, retrieve-based QA)~ However, for NLG tasks (e.g., text generation) or NLU+NLG tasks (e.g., generated QA, Machine Translation), word vector initialization seemingly makes no sense.

**Comment by vince62s** [comment]
xavier initialization has a huge impact on the Transformer performance.
we saw it from day 1 with Sasha when implementing it in OpenNMT-py.
